From ad5c297dfebc2b7c8379c752ad7f2e6d27740893 Mon Sep 17 00:00:00 2001
From: Junming Liu <junming.liu@intel.com>
Date: Thu, 30 Apr 2020 00:42:35 +0800
Subject: [PATCH 1/2] Use the default vbt released with GOP driver

This patch use the default vbt which released along with GOP driver.

Tracked-On: projectacrn/acrn-hypervisor#4365

Signed-off-by: Xiong Zhang <xiong.y.zhang@intel.com>
Signed-off-by: Junming Liu <junming.liu@intel.com>
---
 OvmfPkg/OvmfPkgX64.dsc |  1 +
 OvmfPkg/OvmfPkgX64.fdf |  1 +
 OvmfPkg/Vbt/Vbt.inf    | 17 +++++++++++++++++
 3 files changed, 19 insertions(+)
 create mode 100644 OvmfPkg/Vbt/Vbt.inf

Index: acrn-edk2/OvmfPkg/OvmfPkgX64.dsc
===================================================================
--- acrn-edk2.orig/OvmfPkg/OvmfPkgX64.dsc
+++ acrn-edk2/OvmfPkg/OvmfPkgX64.dsc
@@ -756,6 +756,7 @@
   #OvmfPkg/XenIoPciDxe/XenIoPciDxe.inf
   #OvmfPkg/XenBusDxe/XenBusDxe.inf
   #OvmfPkg/XenPvBlkDxe/XenPvBlkDxe.inf
+  OvmfPkg/Vbt/Vbt.inf
   OvmfPkg/PlatformGopPolicy/PlatformGopPolicy.inf
   MdeModulePkg/Universal/WatchdogTimerDxe/WatchdogTimer.inf
   MdeModulePkg/Universal/MonotonicCounterRuntimeDxe/MonotonicCounterRuntimeDxe.inf
Index: acrn-edk2/OvmfPkg/OvmfPkgX64.fdf
===================================================================
--- acrn-edk2.orig/OvmfPkg/OvmfPkgX64.fdf
+++ acrn-edk2/OvmfPkg/OvmfPkgX64.fdf
@@ -235,6 +235,7 @@ INF  OvmfPkg/GvtGopDxe/GvtGopDxe.inf
 #INF  OvmfPkg/XenIoPciDxe/XenIoPciDxe.inf
 #INF  OvmfPkg/XenBusDxe/XenBusDxe.inf
 #INF  OvmfPkg/XenPvBlkDxe/XenPvBlkDxe.inf
+INF  RuleOverride=CSM OvmfPkg/Vbt/Vbt.inf
 INF  OvmfPkg/PlatformGopPolicy/PlatformGopPolicy.inf
 
 !if $(SECURE_BOOT_ENABLE) == TRUE
Index: acrn-edk2/OvmfPkg/Vbt/Vbt.inf
===================================================================
--- /dev/null
+++ acrn-edk2/OvmfPkg/Vbt/Vbt.inf
@@ -0,0 +1,17 @@
+## @file
+#  Vbt Binary
+#
+#  Copyright (c) 2006 - 2011, Intel Corporation. All rights reserved.<BR>
+#  SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+##
+
+[Defines]
+  INF_VERSION                    = 0x00010005
+  BASE_NAME                      = Vbt
+  FILE_GUID                      = 1647B4F3-3E8A-4FEF-81C8-328ED647AB1A
+  MODULE_TYPE                    = USER_DEFINED
+  VERSION_STRING                 = 1.0
+
+[Binaries]
+  BIN|Vbt.bin|*
